infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/SshKeysProvisioner.java

```diff
@@ -99,12 +101,37 @@ public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
       throws InfrastructureException {
     TracingTags.WORKSPACE_ID.set(identity::getWorkspaceId);
 
+    List<SshPairImpl> vcsSshPairs = getVcsSshPairs(identity);
```

Comment:
I propose to rewrite it in the following way
```suggestion
    List<SshPairImpl> vcsSshPairs = getVcsSshPairs(identity);
    List<SshPairImpl> systemSshPairs = getSystemSshPairs(identity);

    List<SshPairImpl> all = new ArrayList<>(vcsSshPairs);
    all.addAll(systemSshPairs);
    doProvisionSshKeys(all, k8sEnv, identity.getWorkspaceId());

    doProvisionVcsSshConfig(vcsSshPairs, k8sEnv);
    }
... doProvisionVcsSshConfig(... {
        StringBuilder sshConfigData = new StringBuilder();
    for (SshPair sshPair : vcsSshPairs) {
      sshConfigData.append(buildConfig(sshPair.getName()));
    }

    String sshConfigMapName = identity.getWorkspaceId() + SSH_CONFIG_MAP_NAME_SUFFIX;

    ...
    }
```